pandas/core/groupby/groupby.py

```diff
@@ -2075,12 +2075,19 @@ def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                                                      fill_method=fill_method,
                                                      limit=limit, freq=freq,
                                                      axis=axis))
+        if fill_method:
+            new = DataFrameGroupBy(self._obj_with_exclusions,
+                                   grouper=self.grouper)
+            new.obj = getattr(new, fill_method)(limit=limit)
+            new._reset_cache()
+        else:
+            new = self
 
-        filled = getattr(self, fill_method)(limit=limit).drop(
-            self.grouper.names, axis=1)
-        shifted = filled.shift(periods=periods, freq=freq)
+        obj = new.obj.drop(self.grouper.names, axis=1)
+        shifted = new.shift(periods=periods, freq=freq).\
```

Comment:
Side note - I thought this would fail LINTing but regardless we tend to prefer implicit line continuation, so would be better to do `.drop(` on the first line and continue along with the subsequent code on the next line, so long as that all fits